### PR TITLE
soc: npcx: Move ecst to npcx common folder

### PR DESCRIPTION
**Pull request by WealianLiao**
NPCX series ROM code changes the chip basic setting by firmware binary
header for loading the firmware from flash to RAM. Add the following to
improve the ec firmware header setting:

- Move the ECST which generates the firmware binary header to NPCX common
folder. All the following NPCX series chips can use this.
- Add ecst setting option in Kconfig.

Signed-off-by: Yuval Peress <peress@chromium.org>
Signed-off-by: Wealian Liao <WHLIAO@nuvoton.com>

### CODE REVIEW

**Review by sjg20 — APPROVED** [review]

**Review by yperess — APPROVED** [review]

File: soc/arm/nuvoton_npcx/Kconfig [hunk 1]
```diff
@@ -11,6 +11,155 @@ config SOC_FAMILY
...
+	default "quad" if NPCX_HEADER_SPI_READ_MODE_QUAD
+
+choice
+	prompt "Core clock to SPI flash clock ratio"
+	default NPCX_HEADER_CORE_CLOCK_SPI_CLOCK_RATIO_1
+	help
+	  This sets the clock ratio (core clock / SPI clock)
+
+config NPCX_HEADER_CORE_CLOCK_SPI_CLOCK_RATIO_1
+	bool "NPCX SPI clock raiot 1"
```
Comment (keith-zephyr) [inline soc/arm/nuvoton_npcx/Kconfig:93]: Spelling: "raiot" -> "ratio"

File: soc/arm/nuvoton_npcx/Kconfig [hunk 2]
```diff
@@ -11,6 +11,155 @@ config SOC_FAMILY
...
+	help
+	  The SPI flash clock has the same frequency as the core clock.
+
+config NPCX_HEADER_CORE_CLOCK_SPI_CLOCK_RATIO_2
+	bool "NPCX SPI clock raiot 2"
```
Comment (keith-zephyr) [inline soc/arm/nuvoton_npcx/Kconfig:98]: "raiot" -> "ratio"

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: soc/arm/nuvoton_npcx/Kconfig (hunk 1 above)
Comment (WealianLiao) [reply soc/arm/nuvoton_npcx/Kconfig]: Done.

File: soc/arm/nuvoton_npcx/Kconfig (hunk 2 above)
Comment (WealianLiao) [reply soc/arm/nuvoton_npcx/Kconfig]: Done.

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**